### ISSUE TITLE
UDP traffic is "Administratively filtered"

### ISSUE THREAD

**Issue by apinas** [Linutronix/TSN-Testbench#7]
## Description
I am using this repository as an introduction to understand how to setup and configure TSN networks.
When monitoring the _tests/ptm_ output, the reference program UdpHighSent/UdpLowSent counter increments while UdpHighReceived/UdpLowReceived is equal to 0. 
`UdpHighSent=58 | UdpHighReceived=0 | UdpHighRttMin=18446744073709551615 [us] | UdpHighRttMax=0 [us] | UdpHighRttAvg=0.000000 [us] | UdpLowSent=58 | UdpLowReceived=0 | UdpLowRttMin=18446744073709551615 [us] | UdpLowRttMax=0 [us] | UdpLowRttAvg=0.000000 [us] |
`
If I check the output of the mirror, UdpHighSent/UdpLowSent/UdpHighReceived/UdpLowReceived are all zero.
`LldpRttMax=0 [us] | LldpRttAvg=0.000000 [us] | UdpHighSent=0 | UdpHighReceived=0 | UdpHighRttMin=18446744073709551615 [us] | UdpHighRttMax=0 [us] | UdpHighRttAvg=0.000000 [us] | UdpLowSent=0 | UdpLowReceived=0 | UdpLowRttMin=18446744073709551615 [us] | UdpLowRttMax=0 [us] | UdpLowRttAvg=0.000000 [us] `

This traffic can be monitored using Wireshark, where I can see that the reference program sends the UDP datagrams but the mirror answers with ICMP type 3 code 13 (communication administratively filtered):
![image](https://github.com/Linutronix/TSN-Testbench/assets/84238445/f1b15014-f3d5-4826-a66e-35938a5a4122)

When I check the _tests/ptm/flow_cml.sh_ I find the following configuration lines:
```
\# UDP Low: PCP 1 -> Queue 3
ethtool -N ${INTERFACE} flow-type ether vlan 0x2000 m 0x1fff action 3

\# UDP High: PCP 0 -> Queue 3
ethtool -N ${INTERFACE} flow-type ether vlan 0x0000 m 0x1fff action 3
```

I understand that these commands configure a filter so that Ethernet frames that are **VLAN tagged** and have PCP1 and PCP0 are assigned to queue number 3. However, in the frames captured by Wireshark, I can see that the Ethernet frame that embeds the UDP datagram **is not VLAN tagged**. Also, when checking the source code of the test applications, I cannot find any reference to UDP traffic being VLAN tagged.
Could this be because the UDP traffic is "administratively filtered"? I'm worried that I'm missing some NIC configuration.

Thank you in advance!

## Background Information / Reproduction Steps

How can the issue be reproduced?
I cloned the v4.0 tag, and modified the _tests/ptm_ configuration files to match my network interface names, MAC, IP addresses, etc. I ran the mirror on one PC and then launched the reference program in another computer. Both are using an i226-TM NIC. Both cards are directly connected using a CAT6 wire.
I am using a Rocky Linux 9.3 with kernel:
`Linux rockymirror 5.14.0-427.20.1.el9_4.x86_64+rt #1 SMP PREEMPT_RT Fri Jun 7 14:39:50 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux
`

## Checklist

Please provide the following information:

 - [X] Hardware and NIC(s) being used
 - [X] Configuration files being used
 - [X] Linux kernel version being used
 - [X] Linux distribution being used

**Comment by shifty91** [comment]
Hi @apinas,

thanks for your report. The whole application assumes that all traffic classes are **VLAN tagged** (exceptions being PTP and LLDP). The UDP part relies on the existence of a Linux VLAN interface (, as it used standard sockets). That VLAN interface can be created e.g. on Debian in /etc/network/interfaces:

```
auto eth0.100
iface eth0.100 inet ...
```

Or can be created manually with:
```
ip link add link eth0 name eth0.100 type vlan id 100
ip addr add XXX dev eth0.100
```
Hope this helps.

Anyway, this behavior is nowhere documented. I'll add this to the documentation and check for it in the flow scripts.

**Comment by apinas** [comment]
Hi @shifty91,
I imagined that the solution would go that way, but how do I set the PCP field? I saw that there is this _egress-qos-map_ option in the _ip_ command, that allows to map the socket priority SO_PRIORITY to the QoS (PCP field). However, in the application configuration file both UdpLowSocketPriority and UdpHighSocketPriority are set to "3", so both SO_PRIORITY are the same. 
Thank you for your fast answer!
